### DISCUSSION TITLE
Cooling settings for all perimeters (not only external)! infill, etc

### DISCUSSION THREAD

**Discussion by RixYarbrough** [supermerill/SuperSlicer#1506] [Community contribution]
I've been hassling with PrusaSlicer and SuperSlicer for ages now (having the best luck with the latter) to mitigate bulging walls when printing some mechanical parts with PETG. I've concluded the source of the issue is most likely thermal, as some perimeters can be cooled more than others depending on the contiguous size of the layer. In my particular situation I need (or very much prefer) to print external perimeters first, then internal ones. Since there is only a cooling setting for cooling external perimeters independently, in order to cool the internal perimeters at a higher strength (to prevent them from transferring heat energy to the already laid external perimeters, causing them to warp), the only setting I can change is the overall default cooling, meaning the same higher strength cooling will be applied to the infill as well. Since my mechanical parts really only even need walls to assure dimensional accuracy, the infill is the primary structural component. With PETG lower cooling is highly preferable for the infill in order to assure good layer bonding, so powerful cooling only on the perimeters is what I need...

**TLDR**: Additional settings for cooling perimeters, infill, etc separately would be really, really helpful in some situations. I'm no coder so I'm not sure how difficult this could be to implement, but it would be extremely useful.

**Comment by supermerill** [comment]
If you want to make a feature request, you should create an issue instead of opening a discussion.

That said, I won't add yet another cooling thingy until the cooling code is revamped, so it may come in the long term.

**Reply by RixYarbrough** [reply]
Sorry- not well versed in any sort of github community stuff. Thanks for the reply and for your excellent work on this project.
